File: app/src/main/java/closer/vlllage/com/closer/api/models/EventResult.java
package closer.vlllage.com.closer.api.models;

import java.util.Date;
import java.util.List;

import closer.vlllage.com.closer.store.models.Event;

public class EventResult extends ModelResult {
    public List<Double> geo;
    public String name;
    public String about;
    public Date startsAt;
    public Date endsAt;
    public boolean cancelled;
    public String groupId;

    public static Event from(EventResult eventResult) {
        Event event = new Event();
        event.setId(eventResult.id);
        updateFrom(event, eventResult);
        return event;
    }

    public static Event updateFrom(Event event, EventResult eventResult) {
        event.setName(eventResult.name);
        event.setAbout(eventResult.about);
        event.setLatitude(eventResult.geo.get(0));
        event.setLongitude(eventResult.geo.get(1));
        event.setEndsAt(eventResult.endsAt);
        event.setStartsAt(eventResult.startsAt);
        event.setCancelled(eventResult.cancelled);
        event.setGroupId(eventResult.groupId);
        return event;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/api/models/EventResult.kt
package closer.vlllage.com.closer.api.models

import closer.vlllage.com.closer.handler.event.EventReminder
import closer.vlllage.com.closer.store.models.Event
import com.google.gson.annotations.SerializedName
import java.util.*

class EventResult : ModelResult() {
    var geo: List<Double>? = null
    var name: String? = null
    var about: String? = null
    @SerializedName("public")
    val isPublic: Boolean = false
    var startsAt: Date? = null
    var endsAt: Date? = null
    var cancelled: Boolean = false
    var allDay: Boolean = false
    var groupId: String? = null
    var creator: String? = null
    var reminders: List<EventReminder>? = null

    companion object {

        fun from(eventResult: EventResult): Event {
            val event = Event()
            event.id = eventResult.id
            updateFrom(event, eventResult)
            return event
        }

        fun updateFrom(event: Event, eventResult: EventResult): Event {
            event.name = eventResult.name
            event.about = eventResult.about
            event.isPublic = eventResult.isPublic
            event.latitude = eventResult.geo!![0]
            event.longitude = eventResult.geo!![1]
            event.endsAt = eventResult.endsAt
            event.startsAt = eventResult.startsAt
            event.allDay = eventResult.allDay
            event.cancelled = eventResult.cancelled
            event.groupId = eventResult.groupId
            event.creator = eventResult.creator
            event.reminders = eventResult.reminders
            return event
        }
    }
}
